src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -297,24 +303,49 @@ private void InterceptCallAndAdjustArguments(
                     receiverOpt = _factory.Sequence(locals: [], sideEffects: [assignmentToTemp], receiverTemp);
                 }
 
-                arguments = arguments.Insert(0, receiverOpt);
-                receiverOpt = null;
-
-                // CodeGenerator.EmitArguments requires that we have a fully-filled-out argumentRefKindsOpt for any ref/in/out arguments.
-                if (argumentRefKindsOpt.IsDefault && thisRefKind != RefKind.None)
+                if (interceptor.IsExtensionMethod)
                 {
-                    argumentRefKindsOpt = method.Parameters.SelectAsArray(static param => param.RefKind);
-                }
+                    arguments = arguments.Insert(0, receiverOpt);
+                    receiverOpt = null;
 
-                if (!argumentRefKindsOpt.IsDefault)
-                {
-                    argumentRefKindsOpt = argumentRefKindsOpt.Insert(0, thisRefKind);
+                    // CodeGenerator.EmitArguments requires that we have a fully-filled-out argumentRefKindsOpt for any ref/in/out arguments.
+                    if (argumentRefKindsOpt.IsDefault && thisRefKind != RefKind.None)
+                    {
+                        argumentRefKindsOpt = method.Parameters.SelectAsArray(static param => param.RefKind);
+                    }
+
+                    if (!argumentRefKindsOpt.IsDefault)
+                    {
+                        argumentRefKindsOpt = argumentRefKindsOpt.Insert(0, thisRefKind);
+                    }
                 }
             }
 
             method = interceptor;
```

Comment:
>method = interceptor; [](http://example.com/codeflow?start=12&length=21)

Let's say ```interceptor``` is an instance extension and ```method``` is invoked as extension. It looks like we are going to use implementation form of ```interceptor``` to match the signature to ```method```. So far good. But then here we are replacing ```method``` with ```interceptor``` (the declaration form). How are they going to align in the BoundCall that will be created. There would be no receiver and too many arguments in that call. I am failing to connect the dots. What am I missing?

Reply:
We're constructing a busted BoundCall from lowering (lacks receiver, has extra argument) but the `ExtensionMethodReferenceRewriter` hid the problem. I'll add an assertion there. Thanks for catching this